torch/_inductor/kernel/flex_attention.py

```diff
@@ -824,6 +827,177 @@ def flex_attention(
         SPARSE_KV_BLOCK_SIZE,
         mask_graph,
     ) = block_mask
+    if query.get_device().type == "cpu":
```

Comment:
This branch seems to have a lot of code reuse with the following one. I think if we want to have independent branches lets create a seperate `lower_cpu` func and call otherwise I feel like there might be a cleaner way to struxcture this so that the output_logsumexp can just be a check

Reply:
Agree, and have created a seperate `lower_cpu`  function for cpu branch. Thanks.